persist/src/main/java/com/codeforcommunity/database/tableImpl/PostTableDBImpl.java

```diff
@@ -0,0 +1,188 @@
+    // Convert the timestamp to a human-readable string.
+    String time = timestampToString(res.getTimestamp("date_created"));
+    // Return a new PostRecord with the found data. You can either get the column by the index
+    // (so if we said 'SELECT id, author, ...', id would be index 1, author 2, ...) or by column
+    // name.
+    return new PostRecord(
+        res.getInt("id"),
+        res.getString("author"),
+        time,
+        res.getString("title"),
+        res.getInt("clap_count"),
+        res.getString("body"));
+  }
+
+  @Override
+  public PostRecord getById(int id) {
+    PostRecord record = null;
+    try {
+      // Get our database connection.
+      Connection conn = getConnection();
+      // Create our SQL string. This one gets all of the fields of a Post by a given ID.
+      // The '?' allows us to safely insert that variable into the query without having to worry
+      // about escaping any special characters inside.
+      String sql = "SELECT * FROM posts WHERE id = ?;";
+      // A PreparedStatement is the technique that allows us to insert variables by '?'.
+      PreparedStatement stmt = conn.prepareStatement(sql);
+      // Set the first '?' = id. Note how in prepared statements, parameters are not 0-indexed.
+      stmt.setInt(1, id);
+
+      // Get our results. This could also be done in two separate calls;
+      // stmt.execute() and stmt.getResultSet().
+      ResultSet res = stmt.executeQuery();
+      // The next row in the table is queued up by calling ResultSet.next(). If ResultSet.next()
+      // returns false, then there are no more rows (or no rows were found if
+      // this is the first call).
+      if (res.next()) {
+        record = allFieldsResultSetToRecord(res);
+      }
+
+      res.close();
+      stmt.close();
+      conn.close();
+    } catch (SQLException e) {
+      throw new IllegalStateException("There was an issue interacting with the database.", e);
+    }
+    if (record == null) {
+      throw new IllegalArgumentException("No post with ID " + id + " exists.");
+    }
+    return record;
+  }
+
+  @Override
+  public List<PostRecord> getAllPosts() {
+    List<PostRecord> posts = new ArrayList<>();
+    try {
+      Connection conn = getConnection();
+      String sql = "SELECT * FROM posts;";
+      PreparedStatement stmt = conn.prepareStatement(sql);
+      ResultSet res = stmt.executeQuery();
+
+      // Since ResultSet.next() queues up the next row and lets you know if there are any left,
+      // we just iterate through the found records like this.
+      while (res.next()) {
+        posts.add(allFieldsResultSetToRecord(res));
+      }
+
+      res.close();
+      stmt.close();
+      conn.close();
+    } catch (SQLException e) {
+      throw new IllegalStateException("There was an issue interacting with the database.", e);
+    }
+
+    return posts;
+  }
+
+  @Override
+  public boolean postExists(int postId) {
```

Comment:
Is this method checking if a post exists by checking whether a comment exists? Or is the variable supposed to be called postExists and not commentExists?

Reply:
Oops, I thought I answered this one. It checks if a post exists in the post table by ID